cmake/KokkosCore_config.h.in

```diff
@@ -55,45 +54,38 @@
 
 #cmakedefine KOKKOS_IMPL_CUDA_CLANG_WORKAROUND
 
+#cmakedefine KOKKOS_COMPILER_CUDA_VERSION @KOKKOS_COMPILER_CUDA_VERSION@
 
 #cmakedefine KOKKOS_ARCH_NONE
-#cmakedefine KOKKOS_ARCH_AMDAVX
+#cmakedefine KOKKOS_ARCH_SSE42
 #cmakedefine KOKKOS_ARCH_ARMV80
 #cmakedefine KOKKOS_ARCH_ARMV81
-#cmakedefine KOKKOS_ARCH_ARMV8_THUNDERX
-#cmakedefine KOKKOS_ARCH_ARMV8_TX2
-#cmakedefine KOKKOS_ARCH_SSE42
-#cmakedefine KOKKOS_ARCH_WSM
-#cmakedefine KOKKOS_ARCH_SNB
+#cmakedefine KOKKOS_ARCH_AMD_AVX2
 #cmakedefine KOKKOS_ARCH_AVX
-#cmakedefine KOKKOS_ARCH_HSW
-#cmakedefine KOKKOS_ARCH_BDW
 #cmakedefine KOKKOS_ARCH_AVX2
-#cmakedefine KOKKOS_ARCH_SKX
 #cmakedefine KOKKOS_ARCH_AVX512XEON
 #cmakedefine KOKKOS_ARCH_KNC
 #cmakedefine KOKKOS_ARCH_AVX512MIC
-#cmakedefine KOKKOS_ARCH_KNL
-#cmakedefine KOKKOS_ARCH_BGQ
 #cmakedefine KOKKOS_ARCH_POWER7
 #cmakedefine KOKKOS_ARCH_POWER8
 #cmakedefine KOKKOS_ARCH_POWER9
 #cmakedefine KOKKOS_ARCH_KEPLER
+#cmakedefine KOKKOS_ARCH_KEPLER35
 #cmakedefine KOKKOS_ARCH_KEPLER30
 #cmakedefine KOKKOS_ARCH_KEPLER32
-#cmakedefine KOKKOS_ARCH_KEPLER35
```

Comment:
What is the rational for moving that line up?

Reply:
I probably mindlessly did it since Kepler35 is the default if you don't give a number.

Reply:
Why would we allow the user to not pass a number?  I couldn't see the logic that allows that with the generated makefiles.

Reply:
````
ifeq ($(KOKKOS_INTERNAL_USE_CUDA), 1)
  tmp := $(call kokkos_append_header,"\#define KOKKOS_ENABLE_CUDA")
  tmp := $(call kokkos_append_header,"\#define KOKKOS_COMPILER_CUDA_VERSION $(KOKKOS_INTERNAL_COMPILER_NVCC_VERSION)")
endif
````